components/renku_data_services/connected_services/oauth_http.py

```diff
@@ -0,0 +1,611 @@
+"""Module wrapping an oauth library."""
+
+from __future__ import annotations
+
+import time
+from base64 import b64decode, b64encode
+from collections.abc import AsyncGenerator, Callable
+from contextlib import asynccontextmanager
+from enum import StrEnum
+from typing import Any, Protocol
+from urllib.parse import urljoin
+
+import sqlalchemy as sa
+import sqlalchemy.orm as sao
+from authlib.integrations.base_client import InvalidTokenError, OAuthError
+from authlib.integrations.httpx_client.oauth2_client import AsyncOAuth2Client
+from httpx import URL, Response
+from httpx._types import HeaderTypes, QueryParamTypes
+from sqlalchemy.ext.asyncio.session import AsyncSession
+from ulid import ULID
+
+import renku_data_services.connected_services.orm as schemas
+from renku_data_services.app_config import logging
+from renku_data_services.base_api.pagination import PaginationRequest
+from renku_data_services.connected_services import models
+from renku_data_services.connected_services.models import OAuth2Client, OAuth2Connection
+from renku_data_services.connected_services.orm import OAuth2ConnectionORM
+from renku_data_services.connected_services.provider_adapters import (
+    GitHubAdapter,
+    ProviderAdapter,
+    get_provider_adapter,
+)
+from renku_data_services.connected_services.utils import (
+    GitHubProviderType,
+    generate_code_verifier,
+    get_github_provider_type,
+)
+from renku_data_services.errors import errors
+from renku_data_services.repositories.models import OAuth2ClientORM
+from renku_data_services.users.db import APIUser
+from renku_data_services.utils import cryptography as crypt
+
+logger = logging.getLogger(__file__)
+
+
+class OAuthHttpFactoryError(StrEnum):
+    """Errors possible in this module."""
+
+    invalid_connection = "invalid_connection"
+    invalid_user = "invalid_user"
+    no_client = "no_client"
+    no_state = "no_state"
+
+
+class OAuthHttpError(StrEnum):
+    """Errors possible when using the client."""
+
+    oauth_error = "oauth_error"  # nosec: B105
+    unauthorized = "unauthorized"  # nosec: B105
```

Comment:
I just reran the checks for this without the comments and it passed.

```suggestion
    oauth_error = "oauth_error"
    unauthorized = "unauthorized"
```